### ISSUE TITLE
Progbar prints invalid metrics value when using multiple thresholds

### ISSUE THREAD

**Issue by ferryvg** [tensorflow/tensorflow#49161]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- TensorFlow version: 2.4.1
- Python version: 3.8.8

**Describe the current behavior**
Progbar prints mean metric value of all thersholds

**Describe the expected behavior**
Progbar prints first value or value for default threshold (0.5)

**Standalone code to reproduce the issue**
https://github.com/tensorflow/tensorflow/blob/r2.4/tensorflow/python/keras/utils/generic_utils.py#L625
https://colab.research.google.com/drive/1zEa5GB9e8IrufWlHVw2Cc35euc8aXGKy?usp=sharing

**Comment by jvishnuvardhan** [comment]
@ferryvg Can you please share a standalone code to reproduce the issue? Please note that the above standalone code doesn't use `Progabar`. Thanks!

**Comment by ferryvg** [comment]
@jvishnuvardhan
You can just run common training on MIST dataset, for example. You should adds FP/FN/TP/TF metrics to model, set verbose to 1 and compare `Progbar` results with `History` metrics
